docs/source/examples.rst

```diff
@@ -4,6 +4,150 @@
 Examples
 ########
 
+.. _gym_integration:
+
+**************************
+Integration with Gymnasium
+**************************
+
+`OpenAI Gymnasium <https://gymnasium.farama.org/>`_, also known as Gym, is a popular toolkit for developing and comparing reinforcement learning algorithms by provides a standardized interface for environments. Gym offers a variety of environments, from simple classic control tasks like balancing a pole, which is described below in detail, to complex games like Atari and MuJoCo. It even supports creating custom environments, making it a versatile tool for all things reinforcement learning research.
+
+Reinforced-lib on the other hand provides implementations of various reinforcement learning algorithms. It can seamlessly integrate with OpenAI Gym by using the environments provided by Gym as the learning context for the algorithms implemented in Reinforced-lib. This integration allows developers to easily train and evaluate their reinforcement learning models using a wide variety of pre-defined scenarios.
+
+Cart Pole example
+=================
+
+The Cart Pole environment is a classic control task in which the goal is to balance a pole on a cart. The environment is described by a 4-dimensional state space, which consists of the cart's position, the cart's velocity, the pole's angle, and the pole's angular velocity. The agent can take one of two actions: push the cart to the left or push the cart to the right. The episode ends when the pole falls below a certain angle or the cart moves outside of the environment's boundaries. The goal is to keep the pole balanced for as long as possible.
+
+The following example demonstrates how to train a reinforcement learning agent using Reinforced-lib and OpenAI Gym. The agent uses the Deep Q-Network (DQN) algorithm to learn how to balance the pole. The DQN algorithm is implemented in Reinforced-lib and the Cart Pole environment is provided by Gym.
```

Comment:
In the documentation we have "Deep Q-Learning (DQN)", here it is "Deep Q-Network (DQN)". We have to make up our mind and change the description here or there so as not to be misleading.

Reply:
I think the easiest solution (i.e., requiring least explaining) would be to only refer to the "Deep Q-Network (DQN) algorithm".

Reply:
I was reading about it today to find out what the correct form is and I found [this comparison](https://ai.stackexchange.com/a/25918) - I think it best illustrates the difference. Now, however, I don't know whether it is more appropriate to refer to an algorithm or a model in the context of our library?

Reply:
I do not think it is that important. My solution is to leave it as it is already described in agents description which is *Deep Q-learning (DQN)*. I know it is kind of fishy, but it is all over the internet. I was having a headache about it some time in the past :)

Reply:
Moreover, we are explicite explaining the shortcut both here and in the DQN agent class description, so I think it is ok so that we explain to the user what we mean by DQN. Lets not be part of the [Mathematical Inquisition](https://mirror.uncyc.org/wiki/Mathematical_Inquisition)